snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -195,6 +196,9 @@ def _container_run(self, cmd, cwd=None, **kwargs):
         if sh:
             cmd = ['sh', '-c', '{}{}'.format(sh,
                    ' '.join(pipes.quote(arg) for arg in cmd))]
+        # Run commands where file ownership matters as user
+        if os.geteuid() > 0 and (cwd or cmd[0] == 'mkdir'):
```

Comment:
Wait... what's happening here? Turning this if statement into English I get: "If I'm not root, and I either have a cwd or I'm running mkdir, run as \<configured user\>". I'll admit I don't quite understand the problem behind the bug, but I don't see how these three conditionals are related in any way :stuck_out_tongue: . Can you explain that? The comment here isn't helpful.

Reply:
Your translation is correct. If Snapcraft is being run as a user commands inside the container that need to create folders or touch the project folder need to run as the user in the container so that file permissions will match. A counter example is `apt-get` which needs to be run with superuser permissions.

This was the least intrusive option I could think of at the time without modifying callers all over the place. Now I'm only using it for persistent remotes in the cloud, so I'll make it a boolean `user=True`. I find it slightly awkward but maybe it's clearer that way.

Reply:
Btw thanks for questioning this. The code should make sense even without reading the bug and I sometimes don't see the perspective from an unbiased pair of eyes. Much appreciated!